tests/rptest/services/openmessaging_benchmark.py

```diff
@@ -363,10 +366,89 @@ def check_succeed(self, validate_metrics=True):
 
         self._metrics = metrics
 
-        if validate_metrics:
+        if validate_metrics and raise_exceptions:
             OMBSampleConfigurations.validate_metrics(self._metrics,
                                                      self.validator)
 
+        if validate_metrics and not raise_exceptions:
+            is_valid, results = OMBSampleConfigurations.validate_metrics(
+                self._metrics, self.validator, raise_exceptions=False)
+            #if not raise_exceptions:
+            return is_valid, results
+
+    def detect_spikes_by_percentile(self,
+                                    results,
+                                    significant_factor=1.5,
+                                    max_spike_width=1,
+                                    expected_max_latencies=None):
+        def detect_spikes_in_series(latency_series, expected_max):
+            # Sort the series to calculate the median value
+            sorted_series = sorted(latency_series)
+            median_value = sorted_series[len(sorted_series) // 2]
```

Comment:
try to use stuff like numpy which I believe we are already using elsewhere

Reply:
Will be using numpy since we already use in other tests. In this case it is not needed anymore. Logic updated and re-tested.